packages/@aws-cdk/aws-servicecatalog/lib/tag-options.ts

```diff
@@ -1,14 +1,46 @@
+import * as cdk from '@aws-cdk/core';
+import { hashValues } from './private/util';
+import { InputValidator } from './private/validation';
+import { CfnTagOption } from './servicecatalog.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
 /**
- * Defines a Tag Option, which are similar to tags
- * but have multiple values per key.
+ * Defines a set of TagOptions, which are a list of key-value pairs managed in AWS Service Catalog.
+ * It is not an AWS tag, but serves as a template for creating an AWS tag based on the TagOption.
+ * @resource AWS::ServiceCatalog::TagOption
  */
-export class TagOptions {
+export class TagOptions extends cdk.Resource {
   /**
-  * List of CfnTagOption
-  */
-  public readonly tagOptionsMap: { [key: string]: string[] };
+   * Map of underlying TagOption resources.
+   *
+   * @internal
+   */
+  public readonly _tagOptionsMap: { [key: string]: CfnTagOption };
+
+  constructor(scope: Construct, id: string, tagOptions: {[key: string]: string[] }) {
```

Comment:
The cdk linting doesn't like this setup (no interface and no props struct) but I don't particularly see value of having classes that are just pass throughs.  I don't know if making it props with a `tagOptions` field is better since it seems redundant to do

```ts
const tagO = new servicecatalog.TagOptions(this, 'TagOptions, {
  tagOptions : {
    key1: ['v1', 'v2', 'v3'],
  }
});
```
Only if its a little bit of future proofing in case we want to add another field to props then it would be easier to make change.  
I will defer to you though on this.

Reply:
I think linting rules have become more stringent?  Build is failing since you apparently need to disable interface ref in js docs ( I had hoped I could just do it in package.json).

Reply:
Let's just introduce a `TagOptionsProps` interface as the third argument, like you suggested in https://github.com/aws/aws-cdk/pull/18314#discussion_r784495921.

Reply:
that works, leaves more wiggle room to not break things in future should changes happen and we want to modify props.